src/db_adapters/access_keys.rs

```diff
@@ -15,107 +14,71 @@ use crate::{metrics, models};
-                    }
-                    near_primitives::views::ActionView::DeleteKey { public_key } => {
-                        access_keys
-                            .entry((public_key.to_string(), receipt.receiver_id.to_string()))
-                            .and_modify(|existing_access_key| {
-                                existing_access_key.deleted_by_receipt_id =
-                                    Some(receipt.receipt_id.to_string());
-                            })
-                            .or_insert_with(|| models::access_keys::AccessKey {
-                                public_key: public_key.to_string(),
-                                account_id: receipt.receiver_id.to_string(),
-                                created_by_receipt_id: None,
-                                deleted_by_receipt_id: Some(receipt.receipt_id.to_string()),
-                                // this is a workaround to avoid additional struct with optional field
-                                // permission_kind is not supposed to change on delete action
-                                permission_kind: models::enums::AccessKeyPermission::FullAccess,
-                                last_update_block_height: block_height.into(),
-                            });
-                    }
-                    near_indexer::near_primitives::views::ActionView::Transfer { .. } => {
-                        if receipt.receiver_id.len() != 64usize {
-                            continue;
-                        }
-                        if let Ok(public_key_bytes) = hex::decode(receipt.receiver_id.as_ref()) {
-                            if let Ok(public_key) =
-                                near_crypto::ED25519PublicKey::try_from(&public_key_bytes[..])
-                            {
-                                access_keys.insert(
-                                    (near_crypto::PublicKey::from(public_key.clone()).to_string(), receipt.receiver_id.to_string()),
-                                    models::access_keys::AccessKey::from_action_view(
-                                        &near_crypto::PublicKey::from(public_key.clone()),
-                                        &receipt.receiver_id,
-                                        &near_primitives::views::AccessKeyView {
-                                            nonce: 0,
-                                            permission: near_primitives::views::AccessKeyPermissionView::FullAccess
-                                        },
-                                        &receipt.receipt_id,
-                                        block_height,
-                                    ),
-                                );
-                            }
-                        }
-                    }
-                    _ => continue,
+    for state_change in state_changes {
+        if let near_primitives::views::StateChangeCauseView::ReceiptProcessing { receipt_hash } =
```

Comment:
Not sure if we should also be watching for any of the other enums which also contain `receipt_hash`?  (`ActionReceiptProcessingStarted`, `ActionReceiptGasReward`, or `PostponedReceipt`)